cmd/metal-api/internal/auditing/meilisearch.go

```diff
@@ -0,0 +1,119 @@
+package auditing
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/meilisearch/meilisearch-go"
+	"github.com/robfig/cron"
+	"go.uber.org/zap"
+)
+
+type meiliAuditing struct {
+	client           *meilisearch.Client
+	index            *meilisearch.Index
+	log              *zap.SugaredLogger
+	indexPrefix      string
+	rotationInterval Interval
+}
+
+func New(c Config) (Auditing, error) {
+	if !c.Enabled {
+		c.Log.Warn("auditing is disabled, can be enabled by setting --auditing-enabled=true")
+		return &noop{}, nil
```

Comment:
Not sure if there should be a `noop` auditing as it still adds boilerplate to the chain. Maybe we can just add the filter to the chain if enabled and otherwise just don't add to the chain, such that `Enabled` is not field in the config.